Agent panics when calculating NetworkPolicy rule stats

**Describe the bug**
<!--
A clear and concise description of what the bug is.
If you believe this bug is a security issue, please don't use this template and follow our [security guidelines](/SECURITY.md)
-->
Thanks @alokmaurya88 for reporting this issue.
```
I0728 07:10:33.400816       1 reconciler.go:260] Reconciling rule 6a4038b06d3a98a2 of NetworkPolicy K8sNetworkPolicy:demo-77-oblhzg/np-1
I0728 07:10:33.401002       1 reconciler.go:260] Reconciling rule b3c580b7b5de3fc1 of NetworkPolicy K8sNetworkPolicy:demo-77-oblhzg/np-2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x2074ee5]

goroutine 270 [running]:
antrea.io/antrea/pkg/agent/stats.calculateRuleDiff(0xc00cb8a510, 0xc011b7a4b0, 0xc00944dce0, 0x0, 0x1)
	/usr/src/antrea.io/antrea/pkg/agent/stats/collector.go:226 +0x345
antrea.io/antrea/pkg/agent/stats.(*Collector).report(0xc0007c13b0, 0xc00df81280, 0x2, 0x0)
	/usr/src/antrea.io/antrea/pkg/agent/stats/collector.go:170 +0xfc
antrea.io/antrea/pkg/agent/stats.(*Collector).Run(0xc0007c13b0, 0xc0003d2240)
	/usr/src/antrea.io/antrea/pkg/agent/stats/collector.go:92 +0x206
created by main.run
	/usr/src/antrea.io/antrea/cmd/antrea-agent/agent.go:315 +0x2c65
```
There was a bug in `calculateRuleDiff` that it didn't process the case of a new rule of existing networkpolicy correctly when it's stats is zero.
This was introduced when rule level stats was added.

**To Reproduce**
1. Create an Antrea NetworkPolicy or ClusterNetworkPolicy that applies to any Pod, wait for one minute.
2. Append a rule to the policy, make sure no traffic will hit this rule.
3. The agent would panic in one minute.

**Versions:**
 - Antrea version (Docker image tag). v1.0.0-v1.2.0

--- a/pkg/agent/stats/collector.go
+++ b/pkg/agent/stats/collector.go
@@ -214,15 +214,17 @@ func calculateRuleDiff(curStatsMap, lastStatsMap map[types.UID]map[string]*stats
 			}
 		} else {
 			for name, curRuleStats := range curStats {
-				lastRuleStats, ruleStatsExists := lastStats[name]
+				lastRuleStats, lastRuleStatsExists := lastStats[name]
 				// curRuleStats.Bytes < lastRuleStats.Bytes could happen
 				// as rules with same name can be deleted and recreated later.
-				if (!ruleStatsExists || curRuleStats.Bytes < lastRuleStats.Bytes) && curRuleStats.Bytes != 0 {
-					ruleTrafficStats := statsv1alpha1.RuleTrafficStats{
-						Name:         name,
-						TrafficStats: *curRuleStats,
+				if !lastRuleStatsExists || curRuleStats.Bytes < lastRuleStats.Bytes {
+					if curRuleStats.Bytes != 0 {
+						ruleTrafficStats := statsv1alpha1.RuleTrafficStats{
+							Name:         name,
+							TrafficStats: *curRuleStats,
+						}
+						stats = append(stats, ruleTrafficStats)
 					}
-					stats = append(stats, ruleTrafficStats)
 				} else if curRuleStats.Bytes > lastRuleStats.Bytes {
 					ruleTrafficStats := statsv1alpha1.RuleTrafficStats{
 						Name: name,

--- a/pkg/agent/stats/collector_test.go
+++ b/pkg/agent/stats/collector_test.go
@@ -340,7 +340,7 @@ func TestCalculateRuleDiff(t *testing.T) {
 		name              string
 		lastStats         map[types.UID]map[string]*statsv1alpha1.TrafficStats
 		curStats          map[types.UID]map[string]*statsv1alpha1.TrafficStats
-		expectedstatsList []cpv1beta.NetworkPolicyStats
+		expectedStatsList []cpv1beta.NetworkPolicyStats
 	}{
 		{
 			name: "new networkpolicy and existing networkpolicy",
@@ -365,6 +365,11 @@ func TestCalculateRuleDiff(t *testing.T) {
 						Packets:  5,
 						Sessions: 5,
 					},
+					"rule3": {
+						Bytes:    0,
+						Packets:  0,
+						Sessions: 0,
+					},
 				},
 				"uid2": {
 					"rule3": {
@@ -374,7 +379,7 @@ func TestCalculateRuleDiff(t *testing.T) {
 					},
 				},
 			},
-			expectedstatsList: []cpv1beta.NetworkPolicyStats{
+			expectedStatsList: []cpv1beta.NetworkPolicyStats{
 				{
 					NetworkPolicy: cpv1beta.NetworkPolicyReference{UID: "uid1"},
 					RuleTrafficStats: []statsv1alpha1.RuleTrafficStats{
@@ -465,7 +470,7 @@ func TestCalculateRuleDiff(t *testing.T) {
 					},
 				},
 			},
-			expectedstatsList: []cpv1beta.NetworkPolicyStats{},
+			expectedStatsList: []cpv1beta.NetworkPolicyStats{},
 		},
 		{
 			name: "negative statistic",
@@ -499,7 +504,7 @@ func TestCalculateRuleDiff(t *testing.T) {
 					},
 				},
 			},
-			expectedstatsList: []cpv1beta.NetworkPolicyStats{
+			expectedStatsList: []cpv1beta.NetworkPolicyStats{
 				{
 					NetworkPolicy: cpv1beta.NetworkPolicyReference{UID: "uid1"},
 					RuleTrafficStats: []statsv1alpha1.RuleTrafficStats{
@@ -545,12 +550,12 @@ func TestCalculateRuleDiff(t *testing.T) {
 					return v.RuleTrafficStats[i].Name < v.RuleTrafficStats[j].Name
 				})
 			}
-			for _, v := range tt.expectedstatsList {
+			for _, v := range tt.expectedStatsList {
 				sort.SliceStable(v.RuleTrafficStats, func(i, j int) bool {
 					return v.RuleTrafficStats[i].Name < v.RuleTrafficStats[j].Name
 				})
 			}
-			assert.ElementsMatch(t, tt.expectedstatsList, actualMetrics)
+			assert.ElementsMatch(t, tt.expectedStatsList, actualMetrics)
 		})
 	}
 }

